engine/core/src/main/java/org/datacleaner/util/LabelUtils.java

```diff
@@ -102,7 +105,19 @@ public static String getLabel(ComponentJob job, boolean includeDescriptorName, b
         final StringBuilder label = new StringBuilder();
         if (Strings.isNullOrEmpty(jobName)) {
             ComponentDescriptor<?> descriptor = job.getDescriptor();
-            label.append(descriptor.getDisplayName());
+            String baseName = descriptor.getDisplayName();
+            if (ReflectionUtils.is(descriptor.getComponentClass(), HasLabelAdvice.class)) {
+                try {
+                    HasLabelAdvice c = (HasLabelAdvice) descriptor.newInstance();
+                    LifeCycleHelper lch = new LifeCycleHelper((DataCleanerConfiguration) null, (AnalysisJob) null,
+                            false);
+                    lch.assignConfiguredProperties(descriptor, c, job.getConfiguration());
+                    baseName = c.getSuggestedLabel();
```

Comment:
this could return null, so I guess a small addition would be to check that it is not (empty string or) null so that we don't get null-labels